test/openai/test_completion.py

```diff
@@ -449,10 +449,40 @@ def success_metrics(responses, solution, **args):
             "prompt": prompts[0],
             "stop": "###",
         }
-        test_data_sample = test_data[0:5]
+        test_data_sample = test_data[0:3]
         result = oai.ChatCompletion.test(
             test_data_sample, vanilla_config, success_metrics
         )
+        test_data_sample = test_data[3:6]
+        result = oai.ChatCompletion.test(
+            test_data_sample,
+            vanilla_config,
+            success_metrics,
+            use_cache=False,
+            agg_method="median",
+        )
+
+        def my_median(results):
+            return np.median(results)
+
+        def my_average(results):
+            return np.mean(results)
+
+        result = oai.ChatCompletion.test(
+            test_data_sample,
+            vanilla_config,
+            success_metrics,
+            use_cache=False,
+            agg_method=my_median,
+        )
+        result = oai.ChatCompletion.test(
```

Comment:
Is it better to separate the aggregation from testing? I think the test function should store the results. If the user wants to use a new aggregation function, they do not need to test again, but operating on previous results.

Reply:
One of the main purposes of this `test` function is to evaluate the aggregated "performance" of a config on some data in addition to generating responses. The "results" you mentioned in "I think the test function should store the results." are more like responses. If one just wants to get the responses, he/she can use `oai.Completion.create`. In addition, the results are cached.